src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -85,4 +95,12 @@ public void ifImageURLExistsUpdateProductImageURL(MultipartFile multipartFile, P
             product.setImageURL(imageURL);
         }
     }
+
+    private void validateProductForRestoration(UUID productId) {
+        Product product = productRepository.findById(productId).orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "Product with id " + productId + " does not exist."));
```

Comment:
Ya tenemos un método privado para esto je

```suggestion
        Product product = this.findProductByIdOrThrowException(productId);
```